### PR TITLE
62 test cases for review usecases

### PR DESCRIPTION
**Pull request by jkahn506**
Created Test cases for ReviewInteractor and UpdateAverageRatings.

Implemented the simple factory design pattern for Review, to make testing easier.

Created a  static getReviewRepo method in ReviewDatabaseReadWriter to make it easier to get an instance of ReviewDatabaseReadWriter.

Fixed a bug in ReviewInteractor where if the loaded ReviewDatabase was null, it would cause an exception.

### CODE REVIEW

**Review by Brenden-Yiping-Wang — DISMISSED** [review]
The simple factory design pattern plays an important role here!

**Review by Brenden-Yiping-Wang — APPROVED** [review]
The simple factory design pattern still plays an important role here!

**Review by Hclover2003 — APPROVED** [review]
looks good

**Review by YadingF — APPROVED** [review]
ok

File: src/test/java/review/ReviewInteractorTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    RecipeReadWriter recipeReadWriter;
+
+    AverageRatingReadWriter ratingReadWriter;
+
+    AverageRatings currentRatings;
+
+    @BeforeEach
+    void setUp() throws IOException, ClassNotFoundException {
+        readWriter = ReviewDatabaseReadWriter.getReviewRepo();
+        current = readWriter.readFromFile("reviews.sav");
```
[the remark this answers is not in the document]
Comment (jkahn506) [reply src/test/java/review/ReviewInteractorTest.java:30]: Good point.

**Pull request merged**